app/src/main/java/fr/free/nrw/commons/utils/ImageUtils.java

```diff
@@ -130,44 +130,42 @@ private static boolean checkIfImageIsDark(Bitmap bitmap) {
         int bitmapHeight = bitmap.getHeight();
 
         int allPixelsCount = bitmapWidth * bitmapHeight;
-        int[] bitmapPixels = new int[allPixelsCount];
         Timber.d("total %s", Integer.toString(allPixelsCount));
-
-        bitmap.getPixels(bitmapPixels,0,bitmapWidth,0,0,bitmapWidth,bitmapHeight);
         int numberOfBrightPixels = 0;
         int numberOfMediumBrightnessPixels = 0;
-        double brightPixelThreshold = 0.025*allPixelsCount;
-        double mediumBrightPixelThreshold = 0.3*allPixelsCount;
+        double brightPixelThreshold = 0.025 * allPixelsCount;
+        double mediumBrightPixelThreshold = 0.3 * allPixelsCount;
 
-        for (int pixel : bitmapPixels) {
-            int r = Color.red(pixel);
-            int g = Color.green(pixel);
-            int b = Color.blue(pixel);
+        for (int x = 0; x < bitmapWidth; x++) {
+            for (int y = 0; y < bitmapHeight; y++) {
+                int pixel = bitmap.getPixel(x, y);
+                int r = Color.red(pixel);
+                int g = Color.green(pixel);
+                int b = Color.blue(pixel);
 
-            int secondMax = r>g ? r:g;
-            double max = (secondMax>b ? secondMax:b)/255.0;
+                int secondMax = r > g ? r : g;
```

Comment:
I know this is the old code, but nice to sort out while we're going over this

```suggestion
                int maxPixel = Math.max(Math.max(r, g), b);
```

And move the /255.0 into the luminance calc. (do same with min)